Tests/StreamChatTests/Query/Sorting/SortingValue_Tests.swift

```diff
@@ -61,8 +61,141 @@ final class SortingValue_Tests: XCTestCase {
+            makeChannel(cidIndex: 2, lastMessageAtOffset: 3, updatedAtOffset: 14),
+            makeChannel(cidIndex: 3, lastMessageAtOffset: 3, updatedAtOffset: 13),
+            makeChannel(cidIndex: 4, lastMessageAtOffset: 3, updatedAtOffset: 12),
+            makeChannel(cidIndex: 5, lastMessageAtOffset: 5, updatedAtOffset: 11)
+        ]
+        let sortingKeys: [Sorting<ChannelListSortingKey>] = [
+            Sorting(key: .lastMessageAt, isAscending: false),
+            Sorting(key: .updatedAt, isAscending: false)
+        ]
+        let sortValues = sortingKeys.compactMap(\.sortValue)
+        XCTAssertEqual(2, sortValues.count)
+        let runtimeSortResult = channels
+            .sorted(using: sortValues)
+            .map(\.cid.id)
+        
+        let nsArrayChannels = NSArray(array: channels.map { ChannelBoxed(channel: $0) })
+        let sortDescriptors = sortingKeys.compactMap { $0.key.sortDescriptor(isAscending: $0.isAscending) }
+        XCTAssertEqual(2, sortDescriptors.count)
+        
+        let sortDescriptorResult = (nsArrayChannels.sortedArray(using: sortDescriptors) as! [ChannelBoxed]).compactMap(\.id)
+        let expectedIds = ["5", "2", "3", "4", "1"]
+        XCTAssertEqual(expectedIds, sortDescriptorResult, "\(sortDescriptorResult)")
+        XCTAssertEqual(expectedIds, runtimeSortResult, "\(runtimeSortResult)")
+    }
+    
+    func test_sortingValue_channelList_whenAllEqualValues_thenEqualsToNSSortDesciptorSortingAndInitialOrderIsKept() {
+        let channels = [
+            makeChannel(cidIndex: 1, lastMessageAtOffset: 1, updatedAtOffset: 11),
+            makeChannel(cidIndex: 2, lastMessageAtOffset: 1, updatedAtOffset: 11),
+            makeChannel(cidIndex: 3, lastMessageAtOffset: 1, updatedAtOffset: 11),
+            makeChannel(cidIndex: 4, lastMessageAtOffset: 1, updatedAtOffset: 11),
+            makeChannel(cidIndex: 5, lastMessageAtOffset: 1, updatedAtOffset: 11)
+        ]
+        let sortingKeys: [Sorting<ChannelListSortingKey>] = [
+            Sorting(key: .lastMessageAt, isAscending: false),
+            Sorting(key: .updatedAt, isAscending: false)
+        ]
+        let sortValues = sortingKeys.compactMap(\.sortValue)
+        XCTAssertEqual(2, sortValues.count)
+        let runtimeSortResult = channels
+            .sorted(using: sortValues)
+            .map(\.cid.id)
+        
+        let nsArrayChannels = NSArray(array: channels.map { ChannelBoxed(channel: $0) })
+        let sortDescriptors = sortingKeys.compactMap { $0.key.sortDescriptor(isAscending: $0.isAscending) }
+        XCTAssertEqual(2, sortDescriptors.count)
+        
+        let sortDescriptorResult = (nsArrayChannels.sortedArray(using: sortDescriptors) as! [ChannelBoxed]).compactMap(\.id)
+        let expectedIds = ["1", "2", "3", "4", "5"]
+        XCTAssertEqual(expectedIds, sortDescriptorResult, "\(sortDescriptorResult)")
+        XCTAssertEqual(expectedIds, runtimeSortResult, "\(runtimeSortResult)")
+    }
+    
```

Comment:
TODO: Add one test-case for ascending and descending sorting. E.g. sort by unread ascending AND sort by created descending